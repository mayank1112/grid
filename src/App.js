import './App.css';
import { useEffect } from 'react';
import Box from './components/Box/Box';
import axios from 'axios';
import { useState } from 'react';

function App() {
  const [data, setData] = useState('');

  const generateGrid = async () => {
    const response = await axios.get('/index.json');
    setData(response.data.value);
  };

  useEffect(() => {
    generateGrid();
  }, []);

  const onClick = () => {
    console.log('click');
  };

  return (
    <div className="App">
      <Box
        onClick={onClick}
      >{data}</Box>

    </div>
  );
}

export default App;
