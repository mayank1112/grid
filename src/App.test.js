import Enzyme, { mount, shallow } from 'enzyme';
import App from './App.js';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import axios from 'axios';
import { act } from 'react-dom/cjs/react-dom-test-utils.development';
import React from 'react';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

it('<App> component should call hook', () => {
    jest.spyOn(axios, 'get').mockImplementation(() => Promise.resolve({ data: { value: "2" } }));
    jest.spyOn(React, 'useState').mockImplementation(init => ["1", jest.fn()]);

    let wrapper;
    act(() => {
        wrapper = mount(<App />);
        wrapper.update();
    });
    const component = shallow(<App />);
    expect(axios.get).toHaveBeenCalledTimes(1);
    expect(toJson(wrapper)).toMatchSnapshot();
});
