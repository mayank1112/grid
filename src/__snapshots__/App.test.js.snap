// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> component should call hook 1`] = `
<App>
  <div
    className="App"
  >
    <Box
      onClick={[Function]}
    >
      <div
        className="box"
        onClick={[Function]}
      >
        1
      </div>
    </Box>
  </div>
</App>
`;
